packages/ckeditor5-html-support/tests/htmlcomment.js

```diff
@@ -0,0 +1,236 @@
+
+			const paragraph = root.getNodeByPath( [ 0, 0, 0, 0 ] );
+
+			addMarker( '$comment:1', paragraph, 1 );
+			root._setAttribute( '$comment:1', ' comment 1 ' );
+
+			addMarker( '$comment:2', paragraph, 2 );
+			root._setAttribute( '$comment:2', ' comment 2 ' );
+
+			expect( editor.getData() ).to.equal( '<div><div><div><p>F<!-- comment 1 -->o<!-- comment 2 -->o</p></div></div></div>' );
+		} );
+	} );
+
+	describe( 'marker removal post fixer', () => {
+		it( 'should remove all markers and their corresponding $root attributes', () => {
+			editor.setData( '<p><!-- comment 1 -->Foo<!-- comment 2 --></p>' );
+
+			model.change( writer => {
+				writer.remove( writer.createRangeIn( root ) );
+			} );
+
+			expect( editor.getData( { trim: false } ) ).to.equal( '<p>&nbsp;</p>' );
+
+			const rootAttributes = [ ...root.getAttributeKeys() ].filter( attr => attr.startsWith( '$comment' ) );
+
+			expect( rootAttributes ).to.have.length( 0 );
+
+			const markers = [ ...editor.model.markers ].filter( marker => marker.name.startsWith( '$comment' ) );
+
+			expect( markers ).to.have.length( 0 );
+		} );
+
+		// Currently, this test fails. Removing all content from the editor does not remove markers located at $root's boundary.
+		// Since the markers survive, so do comments and their $root attributes.
+		// See https://github.com/ckeditor/ckeditor5/issues/10117.
+		it.skip( 'should remove all markers located at $root\'s boundary and their corresponding $root attributes', () => {
+			editor.setData( '<!-- comment 1 --><p>Foo</p><!-- comment 2 -->' );
+
+			model.change( writer => {
+				writer.remove( writer.createRangeIn( root ) );
+			} );
+
+			// Currently, the returned data is: <p>&nbsp;</p><!-- comment 2 --><!-- comment 1 -->
+			expect( editor.getData( { trim: false } ) ).to.equal( '<p>&nbsp;</p>' );
+
+			const rootAttributes = [ ...root.getAttributeKeys() ].filter( attr => attr.startsWith( '$comment' ) );
+
+			// Currently, there are 2 $root attributes associated with markers.
+			expect( rootAttributes ).to.have.length( 0 );
+
+			const markers = [ ...editor.model.markers ].filter( marker => marker.name.startsWith( '$comment' ) );
+
+			// Currently, there are 2 markers associated with comment nodes.
+			expect( markers ).to.have.length( 0 );
+		} );
+
+		it( 'should remove markers and their corresponding $root attributes representing removed comments only', () => {
+			editor.setData( '<p><!-- comment 1 -->Foo<!-- comment 2 --></p><p><!-- comment 3 -->Foo<!-- comment 4 --></p>' );
+
+			model.change( writer => {
+				const firstParagraph = root.getChild( 0 );
+
+				writer.remove( writer.createRangeOn( firstParagraph ) );
+			} );
+
+			const rootAttributes = [ ...root.getAttributeKeys() ].filter( attr => attr.startsWith( '$comment' ) );
+
+			expect( rootAttributes ).to.have.length( 2 );
+
+			const markers = [ ...editor.model.markers ].filter( marker => marker.name.startsWith( '$comment' ) );
```

Comment:
I'd change all `markers` to `commentMarkers` in these tests.

Reply:
Done.